hail/src/main/scala/is/hail/expr/types/physical/PContainer.scala

```diff
@@ -317,22 +317,38 @@ abstract class PContainer extends PIterable {
       return Code._empty
     }
 
-    val i = mb.newLocal[Long]
+    val i = mb.newLocal[Int]
+    val mod = mb.newLocal[Int]
     Code(
-      i := PContainer.nMissingBytes(sourceType.loadLength(sourceOffset)).toL,
-      Code.whileLoop(i > 0L,
-        (i >= 8L).mux(
+      mod := (sourceType.loadLength(sourceOffset) % const(8)) - const(1),
+      (mod >= const(0)).mux(
+        Code(
+          i := PContainer.nMissingBytes(sourceType.loadLength(sourceOffset)) - const(1),
+          Code.whileLoop(mod >= const(0),
+            Code(
+              sourceType
+                .isElementMissing(sourceOffset, i * const(8) + mod)
```

Comment:
this could obviously be slightly more efficient, but is run max 7 times.

Reply:
cne(false) is unnecessary here, I think?

Reply:
loadBit returns a Code[Boolean]

Reply:
this is `b != False`, which is the same as `b`. Am I reading this totally wrong?

Reply:
Oh you're right, I switched this from loadBit, and didn't change this section. Thanks!